gui/middleware/notifier.py

```diff
@@ -774,30 +774,35 @@ def stripe_first(a, b):
         altroot = 'none' if path else '/mnt'
         mountpoint = path if path else ('/%s' % (z_name, ))
 
-        p1 = self._pipeopen(
-            "zpool create -o cachefile=/data/zfs/zpool.cache "
-            "-o failmode=continue "
-            "-o autoexpand=on "
-            "-O compression=lz4 "
-            "-O aclmode=passthrough -O aclinherit=passthrough "
-            "-f -m %s -o altroot=%s %s %s" % (mountpoint, altroot, z_name, z_vdev))
-        if p1.wait() != 0:
-            error = ", ".join(p1.communicate()[1].split('\n'))
-            raise MiddlewareError('Unable to create the pool: %s' % error)
-
-        # We've our pool, lets retrieve the GUID
-        p1 = self._pipeopen("zpool get guid %s" % z_name)
-        if p1.wait() == 0:
-            line = p1.communicate()[0].split('\n')[1].strip()
-            volume.vol_guid = re.sub('\s+', ' ', line).split(' ')[2]
-            volume.save()
-        else:
-            log.warn("The guid of the pool %s could not be retrieved", z_name)
+        with client as c:
+            lock = c.call("alert.block_source", "VolumeStatus")
```

Comment:
I think the Volume object will be create long before this for 11.2, which is enough for the pool.query to return it as an "existing" pool. Perhaps we should do that on the places calling this method, before creating Volume entry?